core/network/impl/state_protocol_observer_impl.cpp

```diff
@@ -59,10 +59,12 @@ namespace kagome::network {
         if (auto value_res = batch->tryGet(cursor->key().value());
             value_res.has_value()) {
           const auto &value = value_res.value();
-          entry.entries.emplace_back(
-              StateEntry{cursor->key().value(), value.value()});
-          size += entry.entries.back().key.size()
-                  + entry.entries.back().value.size();
+          if (value.has_value()) {
```

Comment:
It guess check of having value should happen before getting value

Reply:
It seems it's value inside value. The first check was inside ```if``` statement

Reply:
optional value inside state key, value pair always would be a cause of confusion. Renamed it to value_opt